### DISCUSSION TITLE
Finalizing blocks appears to be slow when checkpointing large blocks

### DISCUSSION THREAD

**Discussion by jvff** [ZcashFoundation/zebra#3672] [Ideas]
## Motivation

The block synchronizer sometimes restarts when it all of the fanned out extend tip requests time out. When that happens, it can take a few minutes (5-15 minutes) for new blocks to start being downloaded because the request to obtain the block locator from takes a while to be processed by the state service. This happens because the state is busy committing checkpointed finalized blocks.

It would be nice to see if this can be improved to improve the synchronization performance.

## Related Work

PRs:
- temporary fix: #3455

Tickets:
- #3603 
- #3375 
- #3042

**Comment by teor2345** [comment]
If the state is busy committing blocks, then we should speed up the state commit code.

(If the syncer pauses while this happens, this is correct behaviour, and it shows that our backpressure is actually working.)

**Comment by ftm1000** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/3438?planning-poker) with ZenHub @conradoplg @dconnolly @gustavovalverde @jvff @mpguerra @oxarbitrage @upbqdn

**Comment by ftm1000** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/3438?planning-poker) with ZenHub @teor2345

**Comment by teor2345** [comment]
This isn't causing us any issues right now.
